src/Nav.js

```diff
@@ -5,165 +5,262 @@ import ReactDOM from 'react-dom';
 import all from 'react-prop-types/lib/all';
 import warning from 'warning';
 
-import {
-  bsStyles, bsClass as _bsClass, getClassSet, prefix,
-} from './utils/bootstrapUtils';
-import chain from './utils/createChainedFunction';
-import { nextEnabled, PANE, TAB } from './utils/TabUtils';
+import { bsClass, bsStyles, getClassSet, omitBsProps, prefix }
+  from './utils/bootstrapUtils';
+import createChainedFunction from './utils/createChainedFunction';
 import ValidComponentChildren from './utils/ValidComponentChildren';
 
-class Nav extends React.Component {
+// TODO: Should we expose `<NavItem>` as `<Nav.Item>`?
 
-  componentDidUpdate() {
-    if (this._needsRefocus) {
-      let ul = this.refs.ul && ReactDOM.findDOMNode(this.refs.ul);
-      let tabs = ul ? ul.children || [] : [];
-      let tabIdx = this.eventKeys().indexOf(this.getActiveKey());
+// TODO: This `bsStyle` is very unlike the others. Should we rename it?
 
-      this._needsRefocus = false;
+// TODO: `pullRight` and `pullLeft` don't render right outside of `navbar`.
```

Comment:
Not sure that note is right about the navbar...they are simply given the pull-right/left class in the non-navbar instance, there isn't actually an expectation for how that should look in those cases, but i've found them useful regardless.